control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -3428,7 +3633,7 @@ func TestReconcileDeleteEndpoint(t *testing.T) {
 							},
 						}, nil)
 						require.NoError(t, err)
-					case api.ServiceKindTypical:
+					default:
```

Comment:
🤔 It looks like the terminating gateway will fall into the `default` case which means it doesn't matter if the `component` meta data is set on the token. Maybe we then don't need the `switch` case since the behavior is the same regardless of whether the token is for a gateway or not.

Reply:
Cleaned this up